src/NzbDrone.Core/Indexers/Definitions/Anidub.cs

```diff
@@ -0,0 +1,550 @@
+        }
+
+        protected override async Task DoLogin()
+        {
+            UpdateCookies(null, null);
+
+            var requestBuilder = new HttpRequestBuilder(Settings.BaseUrl + "index.php")
+            {
+                LogResponseContent = true,
+                AllowAutoRedirect = true
+            };
+
+            var mainPage = await _httpClient.ExecuteAsync(new HttpRequest(Settings.BaseUrl));
+
+            requestBuilder.Method = Common.Http.HttpMethod.POST;
+            requestBuilder.PostProcess += r => r.RequestTimeout = TimeSpan.FromSeconds(15);
+            requestBuilder.SetCookies(mainPage.GetCookies());
+
+            var authLoginRequest = requestBuilder
+                .AddFormParameter("login_name", Settings.Username)
+                .AddFormParameter("login_password", Settings.Password)
+                .AddFormParameter("login", "submit")
+                .SetHeader("Content-Type", "application/x-www-form-urlencoded")
+                .Build();
+
+            var response = await ExecuteAuth(authLoginRequest);
+
+            if (response.Content != null && !CheckIfLoginNeeded(response))
+            {
+                UpdateCookies(response.GetCookies(), DateTime.Now + TimeSpan.FromDays(30));
+                _logger.Debug("Anidub authentication succeeded");
+            }
+            else
+            {
+                const string ErrorSelector = "#content .berror .berror_c";
+                var parser = new HtmlParser();
+                var document = await parser.ParseDocumentAsync(response.Content);
+                var errorMessage = document.QuerySelector(ErrorSelector).TextContent.Trim();
+                throw new IndexerAuthException("Anidub authentication failed. Error: " + errorMessage);
+            }
+        }
+
+        protected override bool CheckIfLoginNeeded(HttpResponse httpResponse)
+        {
+            if (httpResponse.Content.Contains("index.php?action=logout"))
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        private IndexerCapabilities SetCapabilities()
+        {
+            var caps = new IndexerCapabilities
+            {
+                TvSearchParams = new List<TvSearchParam>
+                {
+                    TvSearchParam.Q, TvSearchParam.Season, TvSearchParam.Ep
+                },
+                MovieSearchParams = new List<MovieSearchParam>
+                {
+                    MovieSearchParam.Q
+                },
+                BookSearchParams = new List<BookSearchParam>
+                {
+                    BookSearchParam.Q
+                },
+                MusicSearchParams = new List<MusicSearchParam>
+                {
+                    MusicSearchParam.Q
+                }
+            };
+
+            caps.Categories.AddCategoryMapping(2, NewznabStandardCategory.TVAnime, "Аниме TV");
```

Comment:
It probably makes sense to order these, I'm good otherwise on this PR

Reply:
@Qstick Fixed